### PR TITLE
Fix crash AssertNotNull

### PR DESCRIPTION
**Pull request by kyunkakata**
## Description
This should fix the issue https://github.com/wix/react-native-ui-lib/issues/2147
I will create a patch from this commit to release a new update to our production app to check whether the crash still occurs or not.

## Changelog
*Add a quick message for our users about this change (include Compoennt name, relevant props and general purpose of the PR)*

### CODE REVIEW

**Review by M-i-k-e-l — COMMENTED** [review]
Hi @kyunkakata,

Thank you very much for the contribution!

Were you able to reproduce this?

File: lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java [hunk 1]
```diff
@@ -14,15 +14,29 @@ public void run() {
     };
 
     public static void runOnUIThread(Runnable runnable) {
-        ReactContextHolder.getContext().runOnUiQueueThread(runnable);
+        /**
+         * When app was open from background,
+         * has case that uiQueueThread was not fully initialized.
+         * Make sure that only call runOnUiQueueThread when getCurrentActivity() is not null.
+         */
+        if (ReactContextHolder.getContext() != null && ReactContextHolder.getContext().getCurrentActivity() != null) {
```
[the remark this answers is not in the document]
Comment (kyunkakata) [reply lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java:22]: I am not sure about that. I have already applied this code to our production app and wait for the patch is working or not.

File: lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java [hunk 2]
```diff
@@ -14,15 +14,29 @@ public void run() {
...
             @Override
             public void run() {
                 userRunnable.run();
-                ReactContextHolder.getContext().runOnNativeModulesQueueThread(sUIUpdateClosure);
+                /**
+                 * When app was open from background,
+                 * has case that uiQueueThread was not fully initialized.
+                 * Make sure that only call runOnNativeModulesQueueThread when getCurrentActivity() is not null.
+                 */
+                if (ReactContextHolder.getContext() != null && ReactContextHolder.getContext().getCurrentActivity() != null) {
```
[the remark this answers is not in the document]
Comment (kyunkakata) [reply lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java:37]: Yeah, From crash logs in Firebase Crashlytics, It happened both from closing the app and re-opening the app from the background state too.
Here is our crash log event when opening the app from the background state:
```
 0 | Tue Jul 26 2022 22:28:17 GMT+0700 (Indochina Time) | session_start
 1 | Tue Jul 26 2022 22:28:17 GMT+0700 (Indochina Time) | screen_view
 2 | Tue Jul 26 2022 22:28:17 GMT+0700 (Indochina Time) | imp_tracking
 3 | Tue Jul 26 2022 22:28:18 GMT+0700 (Indochina Time) | open_notification
 4 | Tue Jul 26 2022 22:28:21 GMT+0700 (Indochina Time) | screen_view
 5 | Tue Jul 26 2022 22:28:21 GMT+0700 (Indochina Time) | screen
 6 | Tue Jul 26 2022 22:28:23 GMT+0700 (Indochina Time) | screen_view
 7 | Tue Jul 26 2022 22:28:23 GMT+0700 (Indochina Time) | screen
 8 | Tue Jul 26 2022 22:28:23 GMT+0700 (Indochina Time) | user_click
 9 | Tue Jul 26 2022 22:28:23 GMT+0700 (Indochina Time) | screen_view
10 | Tue Jul 26 2022 22:28:23 GMT+0700 (Indochina Time) | screen
11 | Tue Jul 26 2022 22:28:25 GMT+0700 (Indochina Time) | screen_view
12 | Tue Jul 26 2022 22:28:25 GMT+0700 (Indochina Time) | screen
13 | Tue Jul 26 2022 22:28:25 GMT+0700 (Indochina Time) | user_click
14 | Tue Jul 26 2022 22:28:25 GMT+0700 (Indochina Time) | screen_view
15 | Tue Jul 26 2022 22:28:25 GMT+0700 (Indochina Time) | screen
16 | Tue Jul 26 2022 22:28:27 GMT+0700 (Indochina Time) | user_click
17 | Tue Jul 26 2022 22:28:28 GMT+0700 (Indochina Time) | screen_view
18 | Tue Jul 26 2022 22:28:28 GMT+0700 (Indochina Time) | screen
19 | Tue Jul 26 2022 22:38:27 GMT+0700 (Indochina Time) | app_did_enter_bg
20 | Wed Jul 27 2022 01:02:36 GMT+0700 (Indochina Time) | session_start
21 | Wed Jul 27 2022 01:02:36 GMT+0700 (Indochina Time) | app_did_enter_bg
22 | Wed Jul 27 2022 01:02:38 GMT+0700 (Indochina Time) | imp_tracking
23 | Wed Jul 27 2022 01:02:38 GMT+0700 (Indochina Time) | screen_view
24 | Wed Jul 27 2022 01:02:38 GMT+0700 (Indochina Time) | screen
25 | Wed Jul 27 2022 01:02:40 GMT+0700 (Indochina Time) | screen_view
26 | Wed Jul 27 2022 01:02:40 GMT+0700 (Indochina Time) | screen
27 | Wed Jul 27 2022 01:02:40 GMT+0700 (Indochina Time) | user_click
28 | Wed Jul 27 2022 01:02:40 GMT+0700 (Indochina Time) | screen_view
29 | Wed Jul 27 2022 01:02:40 GMT+0700 (Indochina Time) | screen
30 | Wed Jul 27 2022 01:02:42 GMT+0700 (Indochina Time) | user_click
31 | Wed Jul 27 2022 01:02:42 GMT+0700 (Indochina Time) | screen_view
32 | Wed Jul 27 2022 01:02:42 GMT+0700 (Indochina Time) | screen
33 | Wed Jul 27 2022 01:08:45 GMT+0700 (Indochina Time) | app_did_enter_bg
34 | Wed Jul 27 2022 08:48:27 GMT+0700 (Indochina Time) | notification_receive
35 | Wed Jul 27 2022 08:50:41 GMT+0700 (Indochina Time) | notification_receive
36 | Wed Jul 27 2022 08:51:06 GMT+0700 (Indochina Time) | notification_receive
37 | Wed Jul 27 2022 09:05:57 GMT+0700 (Indochina Time) | session_start
```

Here is another crash when the app moves to the background state:
```
0 | Wed Jul 27 2022 08:01:06 GMT+0700 (Indochina Time) | notification_receive
 1 | Wed Jul 27 2022 08:01:07 GMT+0700 (Indochina Time) | notification_open
 2 | Wed Jul 27 2022 08:01:07 GMT+0700 (Indochina Time) | session_start
 3 | Wed Jul 27 2022 08:01:07 GMT+0700 (Indochina Time) | screen_view
 4 | Wed Jul 27 2022 08:01:08 GMT+0700 (Indochina Time) | imp_tracking
 5 | Wed Jul 27 2022 08:01:09 GMT+0700 (Indochina Time) | open_notification
 6 | Wed Jul 27 2022 08:01:09 GMT+0700 (Indochina Time) | open_notification
 7 | Wed Jul 27 2022 08:01:12 GMT+0700 (Indochina Time) | screen_view
 8 | Wed Jul 27 2022 08:01:12 GMT+0700 (Indochina Time) | screen
 9 | Wed Jul 27 2022 08:02:04 GMT+0700 (Indochina Time) | imp_tracking
10 | Wed Jul 27 2022 08:02:14 GMT+0700 (Indochina Time) | user_click
11 | Wed Jul 27 2022 08:02:15 GMT+0700 (Indochina Time) | screen_view
12 | Wed Jul 27 2022 08:02:15 GMT+0700 (Indochina Time) | screen
13 | Wed Jul 27 2022 08:02:15 GMT+0700 (Indochina Time) | screen_view
14 | Wed Jul 27 2022 08:02:15 GMT+0700 (Indochina Time) | screen
15 | Wed Jul 27 2022 08:02:22 GMT+0700 (Indochina Time) | imp_tracking
16 | Wed Jul 27 2022 08:02:48 GMT+0700 (Indochina Time) | user_click
17 | Wed Jul 27 2022 08:02:48 GMT+0700 (Indochina Time) | screen_view
18 | Wed Jul 27 2022 08:02:48 GMT+0700 (Indochina Time) | screen
19 | Wed Jul 27 2022 08:02:49 GMT+0700 (Indochina Time) | imp_tracking
20 | Wed Jul 27 2022 08:03:08 GMT+0700 (Indochina Time) | imp_tracking
21 | Wed Jul 27 2022 08:03:40 GMT+0700 (Indochina Time) | screen_view
22 | Wed Jul 27 2022 08:03:40 GMT+0700 (Indochina Time) | screen
23 | Wed Jul 27 2022 08:03:40 GMT+0700 (Indochina Time) | imp_tracking
24 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen_view
25 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen
26 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | user_click
27 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen_view
28 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen
29 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen_view
30 | Wed Jul 27 2022 08:04:10 GMT+0700 (Indochina Time) | screen
31 | Wed Jul 27 2022 08:04:15 GMT+0700 (Indochina Time) | user_click
32 | Wed Jul 27 2022 08:04:16 GMT+0700 (Indochina Time) | screen_view
33 | Wed Jul 27 2022 08:04:16 GMT+0700 (Indochina Time) | screen
34 | Wed Jul 27 2022 08:06:21 GMT+0700 (Indochina Time) | app_did_enter_bg

```

**Comment by kyunkakata** [comment]
> 

Unfortunately, I cannot reproduce this problem in our simulators or our test devices.
Here is a list of devices has that issue:
```
37%
Xiaomi
37%
samsung
6%
vsmart
6%
HUAWEI
14%
Other (4)
```
Android OS:
```
37%
Android 12
31%
Android 11
14%
Android 9
8%
Android 8
8%
Android 10
2%
Android 7
```

File: lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java (hunk 1 above)
[the remark this answers is not in the document]
Comment (M-i-k-e-l) [reply lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java:22]: I think the `getCurrentActivity` tests are not needed, let's remove that along with the comments (the code is clear enough) and we'll merge it.

[the remark this answers is not in the document]
Comment (kyunkakata) [reply lib/android/src/main/java/com/wix/reactnativeuilib/keyboardinput/utils/RuntimeUtils.java]: Ok. I edited it as you recommend. Thank for reviewing

**Review by M-i-k-e-l — APPROVED** [review]

**Pull request merged**